schema/spec.md

```diff
@@ -242,26 +243,28 @@ The type of the resource is `schema`. Any single `schema` is a container for
 one or more `Versions`, which hold the concrete schema documents or schema
 document references.
 
-Any new schema Version that is added to a schema definition SHOULD be backwards
-compatible with all previous Versions of the schema, meaning that a consumer
-using the new schema would be able to understand data encoded using a prior
-Version of the schema. If a new Version introduces a breaking change, it SHOULD
-be registered as a new schema with a new name.
+All Versions of a schema MUST adhere to the semantic rules of the schema's
+`compatibility` attribute. This specification defines "compatibility" to mean
```

Comment:
I think this definition of compatibility should either be dropped, or expanded to be much more precise. (I suspect dropping it would be better, given that we've already said that precise meanings are going to be resource-specific.)

Reply:
Given this is a domain specific spec (schema docs), I think making it more precise might be the better option. Do you have any suggestion for how to tighten it up? Or do you think that each schema language might introduce variants? @clemensv @deissnerk any thoughts?

Reply:
This is where my domain knowledge is insufficient. I'll just say that "backward compatible" is a far fuzzier concept than is normally appreciated, and *absolutely* depends on the precise context.

It's probably worth at least taking a stab at some precise meanings, while acknowledging that we may not get somewhere that we're happy with.

Reply:
Current text:
```
This specification defines "compatibility" to mean that a consumer using one Version of 
the schema would be able to understand data encoded using another Version of the schema.
```
 Does this sound any better?
```
This specification defines "compatibility" for schemas as follows: version A of a schema is
said to be compatible with version B of a schema if any document that adheres to
the rules specified by schema A also adheres to rules specified by schema B.
```

Reply:
Hmm... that's one approach, but it may not be sufficient to really mean that a consumer would be able to understand the data. There can be additional constraints on "if field X exists, a consumer *must* do Y" which make that compatibility hard to define generically.

Reply:
I think the first bullet should be the other way round, if we expect version B to be later than version A. But otherwise, I think that's *at least* better than the current PR :)